src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/Microsoft.AspNetCore.Razor.Utilities.Shared.csproj

```diff
@@ -13,7 +13,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.ObjectPool" />
-    <PackageReference Include="System.Collections.Immutable" />
+    <PackageReference Include="System.Collections.Immutable" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
```

Comment:
There are transport packages which ship this DLL:
- https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-tools/NuGet/Microsoft.NET.Sdk.Razor.SourceGenerators.Transport/
- https://dev.azure.com/dnceng/public/_artifacts/feed/dotnet-tools/NuGet/Microsoft.CodeAnalysis.Razor.Tooling.Internal/

But they are only consumed by the SDK, I think, and missing System.Collections.Immutable shouldn't be a problem there.

Reply:
This wouldn't be a problem even if the packages would ship publicly. SCI with this given version is already provided by the shared framework. That's why NuGet now flags them as unnecessary package references.